src/reanimated2/js-reanimated/JSReanimated.ts

```diff
@@ -101,54 +101,72 @@ export default class JSReanimated {
     }
 
     const sensor: WebSensor = this.initializeSensor(sensorType, interval);
-    let callback;
-    if (sensorType === SensorType.ROTATION) {
-      callback = () => {
-        let [qw, qx, qy, qz] = sensor.quaternion;
-
-        // Android sensors have a different coordinate system than iOS
-        if (this.platform === Platform.WEB_ANDROID) {
-          [qy, qz] = [qz, -qy];
-        }
-
-        // reference: https://stackoverflow.com/questions/5782658/extracting-yaw-from-a-quaternion
-        const yaw = Math.atan2(
-          2.0 * (qy * qz + qw * qx),
-          qw * qw - qx * qx - qy * qy + qz * qz
-        );
-        const pitch = Math.sin(-2.0 * (qx * qz - qw * qy));
-        const roll = Math.atan2(
-          2.0 * (qx * qy + qw * qz),
-          qw * qw + qx * qx - qy * qy - qz * qz
-        );
-        // TODO TYPESCRIPT on web ShareableRef is the value itself so we call it directly
-        (eventHandler as any)({
-          qw,
-          qx,
-          qy,
-          qz,
-          yaw,
-          pitch,
-          roll,
-          interfaceOrientation: 0,
-        });
-      };
-    } else {
-      callback = () => {
-        let { x, y, z } = sensor;
-        [x, y, z] =
-          this.platform === Platform.WEB_ANDROID ? [-x, -y, -z] : [x, y, z];
-        // TODO TYPESCRIPT on web ShareableRef is the value itself so we call it directly
-        (eventHandler as any)({ x, y, z, interfaceOrientation: 0 });
-      };
-    }
-    sensor.addEventListener('reading', callback);
+    sensor.addEventListener(
+      'reading',
+      this.getSensorCallback(sensor, sensorType, eventHandler)
+    );
     sensor.start();
 
     this.sensors.set(this.nextSensorId, sensor);
     return this.nextSensorId++;
   }
 
+  getSensorCallback = (
+    sensor: WebSensor,
+    sensorType: SensorType,
+    eventHandler: ShareableRef<(data: Value3D | ValueRotation) => void>
+  ) => {
+    switch (sensorType) {
+      case SensorType.ACCELEROMETER:
+      case SensorType.GRAVITY:
+        return () => {
+          let { x, y, z } = sensor;
+          [x, y, z] =
+            this.platform === Platform.WEB_ANDROID ? [-x, -y, -z] : [x, y, z];
```

Comment:
I know that you didn't touch this code but could we unify it with the case below?
```suggestion

          // Android sensors have a different coordinate system than iOS
          if (this.platform === Platform.WEB_ANDROID) {
            [x, y, z] = [-x, -y, -z];
          }

```